kubectl-fdb/cmd/cordon.go

```diff
@@ -116,54 +122,111 @@ kubectl fdb cordon -c cluster --node-selector machine=a,disk=fast
-func cordonNode(kubeClient client.Client, cluster *fdbv1beta2.FoundationDBCluster, nodes []string, namespace string, withExclusion bool, wait bool, sleep uint16) error {
-	fmt.Printf("Start to cordon %d nodes\n", len(nodes))
+func cordonNode(cmd *cobra.Command, kubeClient client.Client, inputClusterName string, nodes []string, namespace string, withExclusion bool, wait bool, sleep uint16, clusterLabel string) error {
+	cmd.Printf("Start to cordon %d nodes\n", len(nodes))
 	if len(nodes) == 0 {
 		return nil
 	}
 
-	var processGroups []string
-
+	operationFailed := false
+	var errors []string
 	for _, node := range nodes {
-		var pods corev1.PodList
-		err := kubeClient.List(ctx.Background(), &pods,
-			client.InNamespace(namespace),
-			client.MatchingLabels(cluster.GetMatchLabels()),
-			client.MatchingFieldsSelector{
-				Selector: fields.OneTermEqualSelector("spec.nodeName", node),
-			})
-
+		pods, err := fetchPods(kubeClient, inputClusterName, namespace, node, clusterLabel)
 		if err != nil {
-			return err
+			operationFailed = true
+			errors = append(errors, fmt.Sprintf("Issue fetching Pods running on node: %s. Error: %s\n", node, err))
+			continue
 		}
 
-		for _, pod := range pods.Items {
-			// With the field selector above this shouldn't be required, but it's good to
-			// have a second check.
-			if pod.Spec.NodeName != node {
-				fmt.Printf("Pod: %s is not running on node %s will be ignored\n", pod.Name, node)
+		clusterNames := getClusterNames(cmd, inputClusterName, pods, clusterLabel)
+		for clusterName := range clusterNames {
+			cmd.Printf("Starting operation on %s, node: %s\n", clusterName, node)
+			cluster, err := loadCluster(kubeClient, namespace, clusterName)
+			if err != nil {
+				errors = append(errors, fmt.Sprintf("unable to load cluster: %s, skipping\n", clusterName))
+				operationFailed = true
 				continue
 			}
 
-			processGroup, ok := pod.Labels[cluster.GetProcessGroupIDLabel()]
-			if !ok {
-				fmt.Printf("could not fetch process group ID from Pod: %s\n", pod.Name)
-				continue
+			var processGroups []string
+			for _, pod := range pods.Items {
+				// With the field selector above this shouldn't be required, but it's good to
+				// have a second check.
+				if pod.Spec.NodeName != node {
+					cmd.Printf("Pod: %s is not running on node %s will be ignored\n", pod.Name, node)
+					continue
+				}
+
+				if cluster.ContainsPod(pod) {
+					processGroup, ok := pod.Labels[cluster.GetProcessGroupIDLabel()]
+					if !ok {
+						cmd.Printf("could not fetch process group ID from Pod: %s\n", pod.Name)
+						continue
+					}
+					processGroups = append(processGroups, processGroup)
+				}
+			}
+			err = replaceProcessGroups(kubeClient, cluster.Name, processGroups, namespace, withExclusion, wait, false, true, sleep)
+			if err != nil {
+				operationFailed = true
+				errors = append(errors, fmt.Sprintf("unable to cordon all Pods for cluster %s\n", cluster.Name))
```

Comment:
Any reason not to print the error here directly?

Reply:
mainly just to print all the errors at the end else user have to search the error with other logs. Another thing we can do here is print the error here and all the errors at the end as well.